### DISCUSSION TITLE
Advice for open source projects looking for a design base in React

### DISCUSSION THREAD

**Discussion by adamfowleruk** [vmware/clarity#6286] [General]
Hello all,

At The Herald Project (https://github.com/theheraldproject and https://heraldprox.io ) we're looking to create a backend API and React based web front end for a set of apps. Rather than start from a blank page with a very basic model in Bootstrap and React we're considering adopting Clarity and beginning from a richer, much better designed, starting point.

We will be starting from scratch on the web UI so have no legacy to worry about. What are the Clarity team's recommendations/thoughts on Clarity Core React's readiness for use in a web app that will be released over the next 2-3 months, and if/how to adopt Clarity?

As part of Linux Foundation Public Health (LFPH) our software is Apache 2.0 licensed. Consumers are likely to be public health authority personnel from around the world. Ideally we need any upstream dependencies to require minimum effort to maintain in our code base - we automate everything we can and due to workload, and are staffed by volunteers working in their spare time on pandemic software - we ideally can't spend time debugging minor version updates if they break our code/web UI. (We need to spend more time on disease risk algorithms.)

Thanks in advance for your advice.

Adam.
TSC Chair, The Herald Project, a series of LFPH.

**Comment by gnomeontherun** [comment]
Hi @adamfowleruk! As far as Clarity Core’s readiness, it's used widely with React teams at VMware and we are heavily invested in supporting and expanding Clarity Core’s abilities. We don’t plan for many breaking changes, in fact the whole reason we are going to have to bump a v6 for Clarity Core in the future is to accommodate one relatively minor (but yet still breaking) change to add a new spacing option, and that is basically a search/replace in apps to take in (possibly with automation on our end to support it seamlessly). That is the kind of stability we aim to have in Clarity Core for the coming decades. It is a goal to have no more breaking changes, and if we do that we are fully prepared to ease the migration and minimize the impact on teams. 

Clarity is fully supported by the Clarity team here at VMware as a mission critical dependency for nearly every application, so it is our mandate to ensure high quality code, support, and minimal churn to help all teams be successful with limited effort. Harbor is another open source project using Clarity through VMware (though the Clarity Angular implementation).
